src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2253,6 +2253,104 @@ public function get_test_authorization_header() {
+	 */
+	public function get_test_health_persistent_object_cache() {
+		/**
+		 * Filters the action URL for the persistent object cache health check.
+		 *
+		 * @since 6.1.0
+		 *
+		 * @param string $action_url Learn more link for persistent object cache health check.
+		 */
+		$action_url = apply_filters(
+			'site_status_persistent_object_cache_url',
+			/* translators: Localized Support reference. */
+			__( 'https://wordpress.org/support/article/optimization/#object-caching' )
+		);
+
+		$result = array(
+			'test'        => 'persistent_object_cache',
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => __( 'Performance' ),
+				'color' => 'blue',
+			),
+			'label'       => __( 'A persistent object cache is being used' ),
+			'description' => sprintf(
+				'<p>%s</p>',
+				__( "A persistent object cache makes your site's database more efficient, resulting in faster load times because WordPress can retrieve your site's content and settings much more quickly." )
+			),
+			'actions'     => sprintf(
+				'<p><a href="%s" target="_blank" rel="noopener">%s <span class="screen-reader-text">%s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+				esc_url( $action_url ),
+				__( 'Learn more about persistent object caching.' ),
+				/* translators: Accessibility text. */
+				__( '(opens in a new tab)' )
+			),
+		);
+
+		if ( wp_using_ext_object_cache() ) {
+			return $result;
+		}
+
+		if ( ! $this->should_suggest_persistent_object_cache() ) {
+			$result['label'] = __( 'A persistent object cache is not required' );
+
+			return $result;
+		}
+
+		$available_services = $this->available_object_cache_services();
+
+		$notes = __( 'Your hosting provider can tell you if persistent object caches are able to be enabled on your site.' );
+
+		if ( ! empty( $available_services ) ) {
+			$notes .= ' ' . sprintf(
+				/* translators: Available object caching services. */
+				__( 'Your host appears to support the following object caching services: %s.' ),
+				implode( ', ', $available_services )
+			);
+		}
+
+		/**
+		 * Filters the second paragraph of the health check's description
+		 * when suggesting the use of a persistent object cache.
+		 *
+		 * Hosts may want to replace the notes to recommend their preferred object caching solution.
+		 *
+		 * Plugin authors may want to append notes (not replace) on why object caching is recommended for their plugin.
+		 *
+		 * @since 6.1.0
+		 *
+		 * @param string $notes              The notes appended to the health check description.
+		 * @param array  $available_services The list of available persistent object cache services.
+		 */
+		$notes = apply_filters( 'site_status_persistent_object_cache_notes', $notes, $available_services );
+
+		$result['status']         = 'recommended';
+		$result['label']          = __( 'You should use a persistent object cache' );
+		$result['badge']['color'] = 'orange';
```

Comment:
The badge should not have changing colors based on status, the color in core is used to indicate which area it belongs to. This is also a necessity for accessibility, where a color change alone is not sufficient, and the label would need to change with it if that were the case.

Instead, the `status` should change with the status of the test (which it does here, so this could be easily resolved by removing the badge color change).

Reply:
I didn't know that, so status is enough to get the right color. ok!

Reply:
@manuelRod: Can you update the PR?

Reply:
@tillkruss should we do this first on the plugin level and then on the core level? or should we iterate directly here?

Reply:
@tillkruss @Clorith this is also being discussed in [here](https://github.com/WordPress/wordpress-develop/pull/2894#discussion_r941100596). We should get a decision that will affect both Site Health ( and future ) modules.

Reply:
fixed

Reply:
@manuelRod @tillkruss @Clorith I've opened https://github.com/WordPress/performance/issues/471 and https://github.com/WordPress/performance/pull/472 to fix this in the Performance Lab plugin throughout all Site Health checks.